feature-group-definitions/array-iterators.yml

```diff
@@ -0,0 +1,24 @@
+name: Array iterators
+description: Array iterator methods (usage in for...of loops)
```

Comment:
Well, I think if an interface like `Array` or `URLSearchParams` becomes iterable, it will get these methods. In WebIDL you have iterable declarations:

```idl
interface URLSearchParams {
  iterable<USVString, USVString>;
};
```
(https://webidl.spec.whatwg.org/#js-iterable)

So, in whenever an interface is or becomes iterable, BCD records this with the methods the interface gains. In the case of arrays, this certainly is entries, keys, values, and @@iterator. So that's why I thought of this stuff as a "bundle".

Reply:
Ahh, I had not seen that before. I was basing my understanding on https://tc39.es/ecma262/multipage/control-abstraction-objects.html#sec-iteration, which only talks about `@@iterator`. This is fine.

Reply:
You only need `@@iterator` for for...of loops to work though.

It's probably fine to just lump these together since they've usually been shipped together, but they have different uses. The 3 methods are useful for calling directly, although less useful on arrays than on other container types.

Maybe iterate BCD and look for cases where these 4 API bits haven't shipped together and see if it's rare enough to ignore?

Reply:
> You only need `@@iterator` for for...of loops to work though.

Yes, originally I had a description without "(usage in for...of loops)" but you suggested adding that earlier in this PR :)

Maybe we could use "and" instead of braces to just mean both things equally:

```suggestion
description: Array iterator methods and usage in for...of loops
```

> Maybe iterate BCD and look for cases where these 4 API bits haven't shipped together and see if it's rare enough to ignore?

IIRC, it was a thing for a while that interfaces gained `iterable` support (per WebIDL's definition) and the bits belonging to it usually shipped together. Maybe revisit when we record API compat data where I assume we will encounter this again.

Reply:
Indeed it's the rollout of `iterable` in Web IDL specs and implementations that made this a bit messy.

Looking at the docs again I see that the 3 methods all return array iterators, so there is a connection.

In the description, maybe just name the 3 methods and the symbol, and still point out that this is typically for use with for...of loops. That should be very clear then.

Reply:
```suggestion
description: Array entries(), keys(), values() methods and @@iterator symbol for usage in for...of loops
```